online-banking-cz/src/main/java/com/adyen/checkout/onlinebankingcz/DefaultOnlineBankingCZDelegate.kt

```diff
@@ -0,0 +1,86 @@
+import com.adyen.checkout.components.model.payments.request.OnlineBankingCZPaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.exception.ComponentException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class DefaultOnlineBankingCZDelegate(
+    private val paymentMethod: PaymentMethod,
+    private val paymentMethodFactory: () -> OnlineBankingCZPaymentMethod
+) : OnlineBankingDelegate<OnlineBankingCZPaymentMethod> {
+
+    private val _outputDataFlow = MutableStateFlow<OnlineBankingOutputData?>(null)
+    override val outputDataFlow: Flow<OnlineBankingOutputData?> get() = _outputDataFlow
+
+    private val _componentStateFlow = MutableStateFlow<PaymentComponentState<OnlineBankingCZPaymentMethod>?>(null)
+    override val componentStateFlow: Flow<PaymentComponentState<OnlineBankingCZPaymentMethod>?> = _componentStateFlow
+
+    private val _exceptionFlow: MutableSharedFlow<CheckoutException> = MutableSingleEventSharedFlow()
+    val exceptionFlow: Flow<CheckoutException> = _exceptionFlow
+
+    override fun getIssuers(): List<OnlineBankingModel> {
+        return paymentMethod.issuers?.mapToModel() ?: getLegacyIssuers(paymentMethod.details)
+    }
+
+    override fun getPaymentMethodType(): String {
+        return paymentMethod.type ?: PaymentMethodTypes.UNKNOWN
+    }
+
+    override fun onInputDataChanged(inputData: OnlineBankingInputData) {
+        val outputData = OnlineBankingOutputData(inputData.selectedIssuer)
+
+        _outputDataFlow.tryEmit(outputData)
+
+        createComponentState(outputData)
+    }
+
+    override fun createComponentState(outputData: OnlineBankingOutputData) {
+        val issuerListPaymentMethod = paymentMethodFactory()
+        issuerListPaymentMethod.type = getPaymentMethodType()
+        issuerListPaymentMethod.issuer = outputData.selectedIssuer?.id ?: ""
+
+        val paymentComponentData = PaymentComponentData(paymentMethod = issuerListPaymentMethod)
+
+        val state = PaymentComponentState(paymentComponentData, outputData.isValid, true)
+        _componentStateFlow.tryEmit(state)
+    }
+
+    fun onExceptionHappen(e: CheckoutException) {
+        _exceptionFlow.tryEmit(e)
+    }
+
+    companion object {
+        private val TAG = LogUtil.getTag()
+        private const val TERMS_CONDITIONS_URL = "https://static.payu.com/sites/terms/files/payu_privacy_policy_cs.pdf"
+    }
+
+    @Suppress("ReturnCount")
+    override fun launchOpenPdf(context: Context) {
+        val uri = Uri.parse(TERMS_CONDITIONS_URL)
+        if (OpenPdfUtils.launchNative(context, uri)) return
```

Comment:
The flow here is a bit inefficient. `launchWithCustomTabs` can be called twice in some cases.

`launchNative` will call `launchWithCustomTabs` if the api level is lower than 30. Let's say that fails, thus returns false. Then `launchWithCustomTabs` will be called again on the next line (and it will probably fail again). 
If you flatten this logic, you should be able to prevent this from happening

Reply:
good catch